File: WebApi/1. Models/WorkerModel.cs
namespace PopDb.Models
{
    public enum Gender // not the best of practices to start at 1
    {
        Male = 1,
        Female = 2
    }

    /// <summary></summary>
    public class WorkerModel
    {
        /// <summary></summary>
        public int Id { get; set; }

        /// <summary></summary>
        public int? IsraeliIdentityNumber { get; set; }

        /// <summary></summary>
        public string? FirstName { get; set; }

        /// <summary></summary>
        public string? LastName { get; set; }

        /// <summary></summary>
        public Gender? Gender { get; set; }

        /// <summary></summary>
        public DateTimeOffset? JobStartDate { get; set; }

        /// <summary></summary>
        public DateTimeOffset? JobEndDate { get; set; }

        /// <summary></summary>
        public string? JobDescription { get; set; }

        /// <summary></summary>
        public int? DepartmentId { get; set; }
    }
}

File: WebApi/5. ClientPages/src/models/worker-model.ts
import { Type } from "class-transformer";
import { DepartmentModel } from "./department-model";

export enum Gender
{
  Male = 1,
  Female = 2
}

/**
 * 
 */
export class WorkerModel
{
  /**  */
  id: number;

  /**  */
  israeliIdentityNumber?: number;

  /**  */
  firstName: string;

  /**  */
  lastName: string;

  /**  */
  gender?: Gender;

  /**  */
  @Type(() => Date)
  jobStartDate?: Date;

  /**  */
  @Type(() => Date)
  jobEndDate?: Date;

  /**  */
  jobDescription: string;

  /**  */
  departmentId?: number;

  // @Type(() => DepartmentModel)
  // /**  */
  // department?: DepartmentModel;
}
